$NetBSD: patch-cpphs.cabal,v 1.2 2022/02/11 09:43:46 pho Exp $

Hunk #1:
  Fix build with GHC 9.2.

Hunk #2, #3:
  This is probably a Cabal bug. When a package defines both a library and
  an executable, and they share the same set of modules, Cabal embeds an
  rpath to the build directory to the executable. We don't appreciate that.

--- cpphs.cabal.orig	2001-09-09 01:46:40.000000000 +0000
+++ cpphs.cabal
@@ -52,7 +52,7 @@ Library
       Hs-Source-Dirs: . old
 
     else
-      Build-Depends: time >=1.5 && <1.11
+      Build-Depends: time >=1.5 && <1.12
       Hs-Source-Dirs: . new
 
     Exposed-Modules:
@@ -71,7 +71,7 @@ Library
         TimeCompat
 
 Executable cpphs
-    Build-Depends: base>=3&&<6, directory <1.4, polyparse>=1.13 && <1.14
+    Build-Depends: base>=3&&<6, directory <1.4, polyparse>=1.13 && <1.14, cpphs
     Main-Is: cpphs.hs
 
     if flag(old-locale)
@@ -81,23 +81,9 @@ Executable cpphs
       Hs-Source-Dirs: . old
 
     else
-      Build-Depends: time >=1.5 && <1.11
+      Build-Depends: time >=1.5 && <1.12
       Hs-Source-Dirs: . new
 
-    Other-Modules:
-        Language.Preprocessor.Cpphs
-        Language.Preprocessor.Unlit
-        Language.Preprocessor.Cpphs.CppIfdef
-        Language.Preprocessor.Cpphs.HashDefine
-        Language.Preprocessor.Cpphs.MacroPass
-        Language.Preprocessor.Cpphs.Options
-        Language.Preprocessor.Cpphs.Position
-        Language.Preprocessor.Cpphs.ReadFirst
-        Language.Preprocessor.Cpphs.RunCpphs
-        Language.Preprocessor.Cpphs.SymTab
-        Language.Preprocessor.Cpphs.Tokenise
-        TimeCompat
-
 Source-Repository head
     Type:     git
     Location: https://github.com/hackage-trustees/malcolm-wallace-universe
